src/main.ts

```diff
@@ -73,13 +73,13 @@ export async function addBinToPath(): Promise<boolean> {
     if (!fs.existsSync(gp)) {
       // some of the hosted images have go install but not profile dir
       core.debug(`creating ${gp}`);
-      io.mkdirP(gp);
+      await io.mkdirP(gp);
```

Comment:
Thanks @Frassle!  Could you update the `dist/` file as well so CI passes?  You can build it yourself, or just download the artifact here: https://github.com/actions/setup-go/actions/runs/2051524406

The `dist/` file is what actually gets run when workflows use the action.